x/oracle/abci.go

```diff
@@ -17,77 +18,131 @@ func EndBlocker(ctx sdk.Context, k Keeper) {
 		return
 	}
 
-	actives := k.GetActiveDenoms(ctx)
-	votes := k.CollectVotes(ctx)
+	// Build valid votes counter and winner map over all validators in active set
+	validVotesCounterMap := make(map[string]int64)
+	winnerMap := make(map[string]types.Claim)
+	powerMap := make(map[string]int64)
```

Comment:
Isn't it cleaner just to query the consensuspower of the validator when you need to add a claim to the winnermap? Creating a new data structure seems to be much

Reply:
To query the consensus power when we need to add a claim, we have to pass staking keeper many times until reaching `getPower` function of the each vote. The struct is passed by value so copied every passing. However, if we make powerMap, it will decrease huge copying cost because the map is passed by reference in golang. 

One more reason is we can decrease I/O operation. The staking keeper's `GetValidator` function always requires DB I/O operation and we do so many redundant lookup.

Reply:
I devise some way to simplify getPower mechanism by introducing `ExchangeRateBallot` with `VoteForTally` which contains Power with `ExchangeRateVote`.

Then we don't need to pass ctx, keeper, or map to /internal/types/*

Reply:
Looks great!